src/util/db/sqltransaction.cpp

```diff
@@ -0,0 +1,85 @@
+#include "util/db/sqltransaction.h"
+
+#include <QtDebug>
+
+#include "util/assert.h"
+
+
+namespace {
+    inline
+    bool beginTransaction(QSqlDatabase& database) {
+        if (!database.isOpen()) {
+            // Should only happen during tests
+            qWarning() << "Failed to begin SQL database transaction on"
+                    << database.connectionName();
+            return false;
+        }
+        if (database.transaction()) {
+            qDebug() << "Started new SQL database transaction on"
+                << database.connectionName();
+            return true;
+        } else {
+            qWarning() << "Failed to begin SQL database transaction on"
+                << database.connectionName();
+            return false;
+        }
+    }
+
+} // anonymous namespace
+
+SqlTransaction::SqlTransaction(QSqlDatabase database)
+    : m_database(database),
+      m_active(beginTransaction(m_database)) {
+}
+
+SqlTransaction::SqlTransaction(SqlTransaction&& other)
+    : m_database(std::move(other.m_database)),
```

Comment:
This code is misleading, since it falls back to the copy constructor. 
The move would make sense, if we use a pointer to QSqlDatabase.